homeassistant/components/ambient_station/__init__.py

```diff
@@ -73,11 +70,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             Websocket(entry.data[CONF_APP_KEY], entry.data[CONF_API_KEY]),
         )
         hass.loop.create_task(ambient.ws_connect())
-        hass.data[DOMAIN][entry.entry_id] = ambient
     except WebsocketError as err:
         LOGGER.error("Config entry failed: %s", err)
         raise ConfigEntryNotReady from err
 
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = ambient
```

Comment:
Can make it even shorter:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = ambient
```